main.go

```diff
@@ -2,16 +2,101 @@ package main
 
 import (
 	"fmt"
+	"log"
+	"os"
+	"runtime"
 	"time"
+
+	"github.com/go-logr/logr"
+	"github.com/go-logr/zapr"
+	"github.com/urfave/cli/v2"
+	"github.com/vshn/go-bootstrap/cfg"
+	"github.com/vshn/go-bootstrap/cmd"
+	"github.com/vshn/go-bootstrap/cmd/example"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
 var (
 	// these variables are populated by Goreleaser when releasing
 	version = "unknown"
 	commit  = "-dirty-"
 	date    = time.Now().Format("2006-01-02")
+
+	// TODO: Adjust app name
+	appName     = "go-bootstrap"
+	appLongName = "a generic bootstrapping project"
 )
 
 func main() {
-	fmt.Printf("version %s, commit %s, date %s\n", version, commit, date)
+	err := app().Run(os.Args)
+	if err != nil {
+		log.Fatalf("unable to start %s: %v", appName, err)
+	}
+}
+
+func before(c *cli.Context) error {
+	logger := newLogger(appName, c.Bool("debug"))
+	cmd.SetAppLogger(c, logger)
+	if c.Args().Present() {
+		logger.WithValues(
```

Comment:
You mean to add the version and commit to every log message?
How are the version and commit helping in EFK and Loki?

Reply:
Sorry there were some words missing...

If I create dashboards with error rates from logs I usually graph by version to quickly see if a version has increased error rates after deployment.

Reply:
I see. But I think the `version` should be enough for that case (corresponds to git tag).
Though I imagine having the version key in every log message makes it unreadable for humans, so I'll do something so that the root logging instance only has this field in every message if the log format is JSON.